geoips/plugin_registry.py

```diff
@@ -0,0 +1,294 @@
+                        if plugin_type not in self._registered_plugins:
+                            self._registered_plugins[plugin_type] = {}
+                            self._interface_mapping[plugin_type] = []
+                        for interface in pkg_plugins[plugin_type]:
+                            interface_dict = pkg_plugins[plugin_type][interface]
+                            if interface not in self._registered_plugins[plugin_type]:
+                                self._registered_plugins[plugin_type][
+                                    interface
+                                ] = interface_dict  # NOQA
+                                self._interface_mapping[plugin_type].append(interface)
+                            else:
+                                merge_nested_dicts(
+                                    self._registered_plugins[plugin_type][interface],
+                                    interface_dict,
+                                )
+                except TypeError:
+                    raise PluginRegistryError(f"Failed reading {reg_path}.")
+            self.validate_registry(self._registered_plugins, "all_registered_plugins")
+        return self._registered_plugins
+
+    def get_plugin_info(self, interface, plugin_name):
+        """Find a plugin in the registry and return its info.
+
+        This should remove all plugin loading from the base interfaces and allow us
+        to only load one plugin at a time
+        """
+        plugin_type = self.identify_plugin_type(interface)
+        if isinstance(plugin_name, tuple):
+            return self.registered_plugins[plugin_type][interface][plugin_name[0]][
+                plugin_name[1]
+            ]
+        return self.registered_plugins[plugin_type][interface][plugin_name]
+
+    def get_interface_plugin_info(self, interface):
+        """Find an interface in the registry and return its corresponding info.
+
+        This should remove all plugin loading from the base interfaces and allow us
+        to only load one plugin at a time
+        """
+        plugin_type = self.identify_plugin_type(interface)
+        return self.registered_plugins[plugin_type][interface]
+
+    def get_interface_plugin_names(self, interface):
+        """List all available plugins in the an interface of the registry.
+
+        This should remove all plugin loading from the base interfaces and allow us
+        to only load one plugin at a time
+        """
+        plugin_type = self.identify_plugin_type(interface)
+        return list(self.registered_plugins[plugin_type][interface].keys())
+
+    def list_plugins(self, interface):
+        """List the plugins available for an interface ONLY based on the registries.
+
+        This should not load any plugins, just return info from the registries.
+        """
+        plugin_type = self.identify_plugin_type(interface)
+        return self.registered_plugins[plugin_type][interface]
```

Comment:
Are these relevant to @jsolbrig comment https://github.com/NRLMMD-GEOIPS/geoips/pull/392#discussion_r1376758485 ?

At this point, all of these are just accessing the relevant info directly from the existing "registered_plugins" dictionary, so @jsolbrig comment is no longer applicable?

Reply:
Yes they are. It also expands on it a bit, so we can list all plugins of a given interface, list the names of plugins in a given interface, and also get the information of a plugin specifically. This only reads from the plugin registry.